app/controllers/microservices_engine/v1/data_controller.rb

```diff
@@ -34,18 +34,25 @@ def update
         data = params['content']
         if data.present?
           data.each do |endpoint|
-            existing = Connection.where(object: endpoint['object'])
+            existing = Connection.where(object: endpoint['object']).first
             if existing.present?
               if endpoint['url'].present?
                 # URL exists so we will update as usual
-                existing.update_attributes(name: endpoint['name'], url: endpoint['url'])
+                existing.update_attributes(
+                  name: endpoint.require(:name),
+                  url: endpoint.require(:url)
+                )
               else
                 # URL is blank, thus the endpoint no longer exists
```

Comment:
(@dfaulken) Alright, here we've done to a situation much like the permissions issue we had in round-three awhile ago.

So, if there's no explicit indicator that a service endpoint is being or has been deprecated, I will have to completely remove every existing Connection object and then rebuild them all every time this endpoint receives a request. Filtering out specifically the endpoints that "aren't there" (i.e. comparing every existing Connection object to see if that stuff is still there) is going to take just as much if not more complexity to pull off than just erasing everything

Is that we want? Just wipe the entire connection db and rebuild it from there every time the endpoint gets a request? That's what round-three does for permission on startup, so it's possible. It'll just get... intensive when/if we get a ton of connection objects like how r3 has a ton of permissions. I don't see the app getting to *that* level of objects, but it is entirely possible that it will slow down enough that the app tries to make a request while the controller is trying to rebuild the database. This will cause exceptions across the board.